ted/src/main/java/frc/robot/commands/FeederLaunchNote.java

```diff
@@ -0,0 +1,91 @@
+// ************************************************************
+// Bishop Blanchet Robotics
+// Home of the Cybears
+// FRC - Crescendo - 2024
+// File: FeedNoteCommand.java
+// Intent: Forms a command to feed the note to the shooter or dunker. 
+// ************************************************************
+
+// ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
+
+package frc.robot.commands;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.common.FeederMode;
+import frc.robot.control.Constants;
+import frc.robot.subsystems.FeederSubsystem;
+
+/**
+ * Forms a command to feed the note to the shooter or dunker
+ * Feeder is run until note is detected or timer has expired
+ */
+public class FeederLaunchNote extends Command
+{
+  private FeederSubsystem feeder;
+  private Timer timer = new Timer();
+  private boolean done = false;
+  private FeederMode direction; 
+  
+  /** 
+  * Creates a new feeder command 
+  * 
+  * @param feederSubsystem - the feeder subsystem
+  * @param feederMode - the direction for the feeder
+  */
+  public FeederLaunchNote(FeederSubsystem feederSubsystem, FeederMode feederMode)
+  {
+    this.feeder = feederSubsystem;
+    this.direction = feederMode;
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(feeder);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize()
+  {
+    // set the feeder in the right direction 
+    feeder.setFeederMode(direction);
+    // intentionally *not* setting feeder speed to 0 here. 
+    // If the feeder is currently running, don't stop it. 
+    timer.reset();
+    timer.start();
+    done = false;
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute()
+  {
+    feeder.setFeederSpeed(Constants.feederSpeed);
+    if (timer.hasElapsed(Constants.feederLaunchTimeoutSeconds))
+    {
+      feeder.setAllStop();
+      done = true;
+    }
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted)
+  {
+    // Intentionally *not* setting feeder speed to 0 here. 
+    // the two conditions above (note detected and timer elasped)
```

Comment:
update comment. this command ignores the tof sensor, so doesn't stop when note is detected.

Reply:
updated